lms/djangoapps/courseware/module_render.py

```diff
@@ -182,6 +183,7 @@ def toc_for_course(request, course, active_chapter, active_section, field_data_c
                                      })
             toc_chapters.append({
                 'display_name': chapter.display_name_with_default,
+                'display_id': name_without_spaces,
```

Comment:
I also noticed what @andy-armstrong mentioned and realized what was going on with the focus and selected, but it might be worth revisiting the focus state to be a bit more different than the selected state so it  doesn't look weird, or have the js focus on the selected item instead.